### DISCUSSION TITLE
Schema support

### DISCUSSION THREAD

**Discussion by Rednas83** [picocss/pico#576] [Ideas]
Would it also be possible to add support for schema?

Something like
https://formkit.com/essentials/schema

**Comment by ricpastori** [comment]
I don’t think schema rendering belongs in Pico’s core.

A FormKit-style schema is a JavaScript data-to-DOM abstraction: it defines how an application generates and updates its markup. Pico’s responsibility starts after that step, when semantic HTML already exists and needs styling.

A schema renderer can generate ordinary form elements and use Pico without any integration from Pico itself. Adding schema support here would couple the CSS framework to a particular runtime and data format without improving its CSS or HTML foundation.
